W-19704481-add-Agent-Fabric-authoring-content-mk (#504)
mulesoft/docs-code-builder: modules/ROOT/pages/af-build-agent-networks-in-a-ci-cd-environment.adoc

= Build Agent Networks in a CI/CD Environment

> If you run operations within a CI/CD environment, you can use Anypoint CLI's plugin to set up, create, build, publish, and deploy agent networks.

Reviewer: ```suggestion
Use Anypoint CLI's plugin to set up, create, build, publish, and deploy agent networks if you run operations within a CI/CD environment. 
```

== Before You Begin

> In addition to meeting the Agent Network Prerequisites, you'll need the following software and permissions to use Anypoint CLI to build agent networks.

Reviewer: Unless you add a link, don't capitalize "prerequisites".

Author: This will be a link. All inline links are in progress.

> In addition to meeting the Agent Network Prerequisites, you'll need the following software and permissions to use Anypoint CLI to build agent networks.

Reviewer: ```suggestion
In addition to meeting the Agent Network Prerequisites, you need the following software and permissions to use Anypoint CLI to build agent networks.
```

> * Java 17. Since agent network binaries require Java to build the agent network, make sure you install Java 17 on your machine and set the JAVA_HOME environment variable.

Reviewer: ```suggestion
* Java 17
+
Agent network binaries require Java to build the agent network. Make sure you install Java 17 on your machine and set the JAVA_HOME environment variable.
```

> * Node 20 (v20.19.4). For more information, see link:https://nodejs.org/en/download[Download Node.js].

Reviewer: ```suggestion
* Node 20 (v20.19.4)
+
For more information, see link:https://nodejs.org/en/download[Download Node.js].
```

=== Permissions and Access

Make sure that you have this Runtime Manager permission to manage agent networks with Anypoint CLI.

* Manage Servers

=== Create a Target Space for Your Environment

> To deploy an agent network, you'll need a target space. You can either create a private space for CloudHub 2.0 deployment or (a namespace for Runtime Fabric?).

Reviewer: ```suggestion
To deploy an agent network, you need a target space. You can either create a private space for CloudHub 2.0 deployment or (a namespace for Runtime Fabric?).
```

Reviewer: Recommendation: Use the "TODO" tool to remind yourself about things that are pending.

==== Create a Private Space for CloudHub 2.0 Deployment

. Open Anypoint Runtime Manager.
. Select the environment to use.
. Select **Private Spaces**.
. Select **Create private space**.
. Enter the name agent-network-space (or a name you prefer).
. Select **Create**.
. Select **Create private network**.

=== Set Up Agent Network Gateways for Your Target Space

After you create a target space, set up the ingress and egress gateways for deployment. For more information, see link:https://docs.mulesoft.com/gateway/latest/flex-architecture-basic-deployments[Flex Gateway Deployment Models]. (Runtime Fabric instrux?)

Open a terminal window and run this command:

----
$ anypoint-cli-agent-fabric-plugin agent-network setup gateways --target-space <space>
----

As a result, the command creates agent-network-ingress and agent-network-egress gateways in the selected target space.

== Default Flags and Authentication

For information on default flags that apply to all CLI commands, see link:https://docs.mulesoft.com/anypoint-cli/latest/#default-options[Default Flags].

> You can authenticate to Anypoint Platform through the command line by running the anypoint-cli-agent-fabric-plugin command with a combination of parameters.

Reviewer: ```suggestion
You can authenticate to Anypoint Platform through the command line by running the `anypoint-cli-agent-fabric-plugin` command with a combination of parameters.
```

> Or, you can set a combination of environment variables before running the anypoint-cli-agent-fabric-plugin command. For more information, see link:https://docs.mulesoft.com/anypoint-cli/latest/auth#command-line-parameters[Command-Line Parameters and Environment Variables].

Reviewer: ```suggestion
Or, you can set a combination of environment variables before running the `anypoint-cli-agent-fabric-plugin` command. For more information, see link:https://docs.mulesoft.com/anypoint-cli/latest/auth#command-line-parameters[Command-Line Parameters and Environment Variables].
```

----

As a result, the command builds the agent network application and asset definitions.

For more information about the command and available flags, run:

----
$ anypoint-cli-agent-fabric-plugin agent-network project build --help
----

== Publish Your Agent Network to Anypoint Exchange

=== Permissions and Access

Make sure you have this Anypoint Exchange permission to publish your agent network assets successfully. For more information, see link:https://docs.mulesoft.com/access-management/permissions-by-product#exchange[Permissions - Exchange].

* Exchange Contributor

=== Command

----
$ anypoint-cli-agent-fabric-plugin agent-network project publish
----

As a result, the command provides the links and details for all the created assets.

For more information about the command and available flags, run:

----
$ anypoint-cli-agent-fabric-plugin agent-network project publish --help
----

After you publish your assets, discover them in **Agents & Tools** in Exchange to get a comprehensive view of your integration, including agent and MCP server assets. Exchange automatically tags brokers for easy identification.

== Deploy Your Agent Network

> Make sure that you've created a target space and set up gateways. For more information, see Before You Begin.

Reviewer: ```suggestion
Make sure that you create a target space and set up gateways. For more information, see Before You Begin.
```
